flang/lib/Lower/CallInterface.cpp

```diff
@@ -1055,15 +1059,24 @@ class Fortran::lower::CallInterfaceImpl {
       const DummyCharacteristics *characteristics,
       const Fortran::evaluate::characteristics::DummyProcedure &proc,
       const FortranEntity &entity) {
-    if (proc.attrs.test(
+    if (!interface.converter.getLoweringOptions().getLowerToHighLevelFIR() &&
```

Comment:
Should this be removed since HLFIR is by default now?

Reply:
No, we need to keep these until the non HLFIR lowering is removed (which should happen early next year). The point is to give a fallback for a while if users face bugs with the HLFIR switch. Sorry for the inconvenience.